### PR TITLE
add promise import instructions + error handling

### PR DESCRIPTION
**Pull request by mekarpeles**
Fix promise item import error handling

Prereq to #7216

### CODE REVIEW

File: scripts/promise_batch_imports.py [hunk 1]
```diff
@@ -21,19 +41,20 @@ def map_book_to_olbook(book, promise_id):
     asin_is_isbn_10 = book.get('ASIN') and book.get('ASIN')[0].isdigit()
     publish_date = book['ProductJSON'].get('PublicationDate')
     title = book['ProductJSON'].get('Title')
+    isbn = book.get('ISBN', ' ')
     olbook = {
         'local_id': [f"urn:bwbsku:{book['BookSKUB']}"],
         'identifiers': {
             **({'amazon': [book.get('ASIN')]} if not asin_is_isbn_10 else {}),
             **(
                 {'better_world_books': [book.get('ISBN')]}
```
Comment (mekarpeles) [inline scripts/promise_batch_imports.py:50]:
```suggestion
                {'better_world_books': [isbn]}
```

File: scripts/promise_batch_imports.py [hunk 2]
```diff
@@ -21,19 +41,20 @@ def map_book_to_olbook(book, promise_id):
     asin_is_isbn_10 = book.get('ASIN') and book.get('ASIN')[0].isdigit()
     publish_date = book['ProductJSON'].get('PublicationDate')
     title = book['ProductJSON'].get('Title')
+    isbn = book.get('ISBN', ' ')
```
Comment (mekarpeles) [inline scripts/promise_batch_imports.py:44]:
```suggestion
    isbn = book.get('ISBN') or ' '
```

File: scripts/promise_batch_imports.py [hunk 3]
```diff
@@ -52,7 +69,11 @@ def map_book_to_olbook(book, promise_id):
 def batch_import(promise_id):
     url = "https://archive.org/download/"
     date = promise_id.split("_")[-1]
-    books = requests.get(f"{url}{promise_id}/DailyPallets__{date}.json").json()
+    try:
+        books = requests.get(f"{url}{promise_id}/DailyPallets__{date}.json").json()
+    except MissingDailyPalletsJSON as e:
```
Comment (mekarpeles) [inline scripts/promise_batch_imports.py:74]: needs to be raw Exception and we need to somehow account or handle as something more specific

Comment (mekarpeles) [reply scripts/promise_batch_imports.py:74]: @cdrini if you have any bandwidth to advise with this one, I'm trying to push forward with the cron portion.

**Comment by mekarpeles** [comment]
Processed up to `bwb_daily_pallets_2022-12-20`

**Comment by mekarpeles** [comment]
@ximm can you help -- What's the ES query to fetch items which have been created in the last two months?

File: scripts/promise_batch_imports.py [hunk 4]
```diff
@@ -59,23 +72,29 @@ def batch_import(promise_id):
...
     q = "collection%3Abookdonationsfrombetterworldbooks+identifier%3Abwb_daily_pallets_*"
+    if start_date:
+        q += f"+publicdate%3A[{start_date} TO {end_date}]"
     sorts = "sort%5B%5D=addeddate+desc&sort%5B%5D=&sort%5B%5D="
     fields = "fl%5B%5D=identifier"
     rows = 5000
-    r = requests.get(f"{url}?q={q}&{fields}&{sorts}&rows={rows}&page=1&output=json")
-    return [d['identifier'] for d in r.json()['response']['docs']]
+    url = f"{base}?q={q}&{fields}&{sorts}&rows={rows}&page=1&output=json"
+    return url
```
Comment (cdrini) [inline scripts/promise_batch_imports.py:90]: Should switch to using a dictionary for better legibility

**Review by cdrini — APPROVED** [review]
Merging despite comment since blocking today's deploy.

File: scripts/promise_batch_imports.py (hunk 3 above)
Comment (cdrini) [reply scripts/promise_batch_imports.py]: Dropping this error catch for now.

**Pull request merged**